src/js/selection/SelectionManager.js

```diff
@@ -101,14 +107,36 @@
 
     this.pastePixels_(frame, pixels);
 
-    $.publish(Events.PISKEL_SAVE_STATE, {
-      type : pskl.service.HistoryService.REPLAY,
-      scope : this,
-      replay : {
-        type : SELECTION_REPLAY.PASTE,
-        pixels : JSON.parse(JSON.stringify(pixels.slice(0)))
+    // Offset the pasted selection from the original location.
+    var tool = pskl.app.drawingController.currentToolBehavior;
+    var isSelectionTool = tool instanceof pskl.tools.drawing.selection.BaseSelect;
+    if (isSelectionTool) {
+      var maxHeight = frame.width - 1;
+      var maxWidth = frame.height - 1;
+      var xCoordinate = pixels[pixels.length - 1].col;
+      var yCoordinate = pixels[pixels.length - 1].row;
```

Comment:
When using the lasso tool, the 'last' pixel won't necessarily represent the lower-right boundary of your selection.